docs/reading-mets.rst

```diff
@@ -18,27 +18,82 @@ metsrw supports reading METS files from disk, from strings, or from lxml_
     mets = metsrw.METSDocument.fromstring(mets_str)
 
     # From an lxml object
-    tree = lxml.etree.fromfile('path/to/file')
+    tree = lxml.etree.fromfile("../fixtures/complete_mets.xml")
     mets = metsrw.METSDocument.fromtree(tree)
 
 
 Accessing METS Data
 -------------------
 
 To retrieve an :class:`metsrw.FSEntry`, use the
-:func:`~metsrw.METSDocument.get_file` method.
+:meth:`~metsrw.METSDocument.get_file` or
+:meth:`~metsrw.METSDocument.all_files` methods.
 
-.. code-block:: python
+.. doctest::
 
-    mets = metsrw.METSDocument()
-    file_uuid = str(uuid.uuid4())
-    file_1 = metsrw.FSEntry(
-        label="hello.pdf", path="test/hello.pdf", type="Item",
-        file_uuid=file_uuid)
-    mets.append_file(file_1)
+    >>> mets = metsrw.METSDocument()
+    >>> file_uuid = str(uuid.uuid4())
+    >>> file_1 = metsrw.FSEntry(
+    ...     label="hello.pdf", path="test/hello.pdf", type="Item",
+    ...     file_uuid=file_uuid)
+    >>> mets.append_file(file_1)
 
-    # Returns file_1
-    mets.get_file(file_uuid=file_uuid)
+    >>> mets.get_file(file_uuid=file_uuid)
+    FSEntry(type='Item', path='test/hello.pdf', use='original', ...)
+
+    >>> mets.all_files()
+    {FSEntry(type='Item', path='test/hello.pdf', use='original', ...)}
+
+    # Currently, filtering files can only be done via iteration
+    >>> [entry for entry in mets.all_files() if entry.use == "original"]
+    [FSEntry(type='Item', path='test/hello.pdf', use='original', ...)]
+
+
+`amdSec` and `dmdSec` data is accessible via the
+:attr:`~metsrw.FSEntry.amdsecs` and :attr:`~metsrw.FSEntry.dmdsecs`
+properties.
+
+.. doctest::
+
+    >>> mets = metsrw.METSDocument.fromfile('../fixtures/complete_mets.xml')
+    >>> fsentry = mets.get_file(file_uuid="ab5c67fc-8f80-4e46-9f20-8d5ae29c43f2")
+    >>> amdsec1 = fsentry.amdsecs[0]
+    >>> [section for section in amdsec1.subsections if section.subsection == 'techMD']
```

Comment:
@replaceafill  should cover this use case:
```
Use the METS tree to get the <amdSec> from a file entry:
# entry is a FSEntry object
amdSec = mets.tree.find(
    'mets:amdSec[@ID="{}"]'.format(entry.admids[0]),
    namespaces={...},
)
```

I think it's probably better to just get the PREMIS data directly, so I've added the note below.

Reply:
Ah! Good suggestion. Thank you @cole!